internal/assertion_state.ts

```diff
@@ -69,26 +142,56 @@ export class AssertionState {
    * import { AssertionState } from "@std/internal";
    *
    * const assertionState = new AssertionState();
-   * if (assertionState.checkAssertionErrorStateAndReset()) {
+   * if (assertionState.checkAssertionErrorState()) {
    *   // throw AssertionError("");
    * }
    * ```
    */
-  checkAssertionErrorStateAndReset(): boolean {
-    const result = this.#state.assertionCheck &&
-      !this.#state.assertionTriggered;
-
-    this.#resetAssertionState();
-
-    return result;
+  checkAssertionErrorState(): boolean {
+    return this.#state.assertionCheck && !this.#state.assertionTriggered;
   }
 
-  #resetAssertionState(): void {
+  /**
+   * Reset all assertion state when every test suite function ran completely.
+   *
+   * @example Usage
+   * ```ts ignore
+   * import { AssertionState } from "@std/internal";
+   *
+   * const assertionState = new AssertionState();
+   * assertionState.resetAssertionState();
+   * ```
+   */
+  resetAssertionState(): void {
     this.#state = {
+      assertionCount: undefined,
       assertionCheck: false,
       assertionTriggered: false,
+      assertionTriggeredCount: 0,
     };
   }
+
+  /**
+   * Check Assertion called state, if `#state.assertionCount` is set to a number value, but
+   * `#state.assertionTriggeredCount` is less then it, then should throw an assertion error.
+   *
+   * @returns a boolean value, that the test suite is satisfied with the check. If not,
+   * it should throw an AssertionError.
+   *
+   * @example Usage
+   * ```ts ignore
+   * import { AssertionState } from "@std/internal";
+   *
+   * const assertionState = new AssertionState();
+   * if (assertionState.checkAssertionCountSatisfied()) {
+   *   // throw AssertionError("");
+   * }
+   * ```
+   */
+  checkAssertionCountSatisfied(): boolean {
+    return this.#state.assertionCount !== undefined &&
+      this.#state.assertionCount > this.#state.assertionTriggeredCount;
```

Comment:
It seems in jest, `expect.assertions(N)` call expects exactly `N` assertions are made. For example, the below test file fails with the error message `Expected one assertion to be called but received two assertion calls.`:

```js
test("test", () => {
  expect.assertions(1);
  expect(1).toBe(1);
  expect(2).toBe(2);
});
```

Reply:
I'll check this later.

Reply:
Thanks, I've updated.